apps/masters/views.py

```diff
@@ -525,3 +526,174 @@ def create(self, request, *args, **kwargs):
 
     def update(self, request, *args, **kwargs):
         return update_instance(self, request, *args, **kwargs)
+
+#===============================================PDF_creation================================
+from reportlab.platypus import SimpleDocTemplate, Paragraph, Spacer
+from reportlab.lib.styles import getSampleStyleSheet,ParagraphStyle
+from apps.masters.utils.table_defination import create_table_1, create_table_2, create_table_3, create_table_4, create_table_5, create_table_6
+from config.utils_methods import convert_amount_to_words, extract_product_data, format_phone_number,send_pdf_via_email, get_related_data, list_all_objects, create_instance, update_instance, build_response
+from django.http import Http404
+from apps.customer.models import CustomerAddresses
+from .utils.docs_variables import doc_data
+from reportlab.lib.pagesizes import inch
+
+class DocumentGeneratorView(APIView):
+    def post(self, request, **kwargs):
+        """ Retrieves a sale order and its related data. """
+
+        try:
+            flag = request.data.get('flag')
+            pk = kwargs.get('pk')
+            document_type = kwargs.get('document_type')
+
+            # Get the relevant data from the doc_data dictionary
+            model_data = doc_data.get(document_type)
+            if model_data:
+                Model = model_data.get('Model')
+                Serializer = model_data.get('Serializer')
+                Item_Model = model_data.get('Item_Model')
+                Items_Serializer = model_data.get('Items_Serializer')
+                Item_Model_PK = model_data.get('Item_Model_PK')
+                Related_Model = model_data.get('Related_Model')
+                Related_Serializer = model_data.get('Related_Serializer')
+                Related_filter_field = model_data.get('Related_filter_field')
+                number_lbl = model_data.get('number_lbl')
+                date_lbl = model_data.get('date_lbl')
+                Doc_Header = model_data.get('Doc_Header')
+                net_lbl = model_data.get('net_lbl')
+                number_value = model_data.get('number_value')
+                date_value = model_data.get('date_value')
```

Comment:
I see some variables in camelCase and some in snake_case, please change variables to snake_case

Reply:
changed